std/experimental/allocator/building_blocks/allocator_list.d

```diff
@@ -196,6 +196,57 @@ struct AllocatorList(Factory, BookkeepingAllocator = GCAllocator)
         return null;
     }
 
+    /**
+    Allocate a block of size `s` with alignment `a`. First tries to allocate
+    from the existing list of already-created allocators. If neither can
+    satisfy the request, creates a new allocator by calling `make(s + a - 1)`
+    and delegates the request to it. However, if the allocation fresh off a
+    newly created allocator fails, subsequent calls to `alignedAllocate`
+    will not cause more calls to `make`.
+    */
+    static if (hasMember!(Allocator, "alignedAllocate"))
+    void[] alignedAllocate(size_t s, uint theAlignment)
+    {
+        import std.algorithm.comparison : max;
+        import core.checkedint : adds;
+
+        for (auto p = &root, n = *p; n; p = &n.next, n = *p)
+        {
+            auto result = n.alignedAllocate(s, theAlignment);
+            if (result.length != s) continue;
+            // Bring to front if not already
+            if (root != n)
+            {
+                *p = n.next;
+                n.next = root;
+                root = n;
+            }
+            return result;
+        }
+        // Can't allocate from the current pool. Check if we just added a new
+        // allocator, in that case it won't do any good to add yet another.
+        if (root && root.empty == Ternary.yes)
+        {
+            // no can do
+            return null;
+        }
+
+        bool overflow = false;
+        size_t maxSize = adds(s - 1, cast(size_t) theAlignment, overflow);
```

Comment:
What if `s` is `0`? Would that be an issue, or would this overflow and thus return `null`? Maybe add a check at function entry?

Reply:
Also, could you add two tests that allocate `0` and `size_t.max`?

Reply:
The allocation requests are passed to `Allocator` which has to correctly treat these cases.

Reply:
Also, added test for size `0`, but for `size_t.max`, the assert would trigger if I add a test.